### PR TITLE
Delegate #transaction to ActiveRecord::Base

### PR DESCRIPTION
**Pull request by thewatts**
### Summary

For our project, we're using some custom database connection logic/pooling, along with multiple databases.

Because of this - it means that we're no longer able to reach for `::ActiveRecord::Base` to wrap transactions.

Instead, we have to use separate classes, ex:

```ruby
ClientApplicationRecord.transaction do
  # do stuff with client connections
end

ApplicationRecord.transaction do
  # do stuff with application connections
end
```

Delegating `#transaction` to `::ActiveRecord::Base` allows us to safely override the delegation for Client specific forms.

```ruby
class ClientForm < YAAF::Form
  delegate :transaction, to: ClientApplicationRecord
end
```

as opposed to having to override the entire `save_in_transaction`
method:

```ruby
class ClientForm < YAAF::Form
  # lots of duplicated logic that may change in future gem versions
  def save_in_transaction(options)
    ::ClientApplicationRecord.transaction do
      run_callbacks :save do
        save_models(options)
      end
    end
  rescue => e
    handle_transaction_rollback(e)
  end
end
```

I went back and forth on whether or not to just make a private `#database_connection_class` method, but doing so could mean that a future committer refactor/remove the change. Making it public means that it can be tested (as it's poor form to directly test private methods), which adds a bit of friction for removal.

As for a _public_ `#database_connection_class` method, `Form`'s _written_ public API is already so small, that having this method hang out in the file with `#save` and `#save!` just didn't quite feel right.

### Other Information
The test included is a little bogus, in that it only tests that `Form` responds to `#transaction`. [Shoulda-Matchers offers a `delegate_method` matcher](https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/independent/delegate_method_matcher.rb#L49) that could be used, but I didn't know if an additional dependency was wanted, especially for such a small use case.

I'm more than happy to adjust the spec as needed.

### CODE REVIEW

File: spec/yaaf/form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+RSpec.describe YAAF::Form do
+  describe '#transaction' do
+    it 'responds to the transaction method' do
```
Comment (santib) [inline spec/yaaf/form_spec.rb:3]: Should we test that the `transaction` method can be delegated to something else?

For example 
```ruby
expect(User).to receive(:transaction).and_call_original
```

**Review by santib — APPROVED** [review]
I like the approach you took 💯 Just left one question regarding tests

File: spec/yaaf/form_spec.rb (hunk 1 above)
Comment (thewatts) [reply spec/yaaf/form_spec.rb:3]: How about something like:

```ruby
it 'is delegated to ActiveRecord::Base' do
  form = YAAF::Form.new

  expect(ActiveRecord::Base).to receive(:transaction).and_call_original

  form.transaction { User.count }
end
```

Comment (santib) [reply spec/yaaf/form_spec.rb:3]: but there you aren't testing that it's possible to override who receives the call to `transaction`, right?

Comment (thewatts) [reply spec/yaaf/form_spec.rb:3]: AHH, I see. Adding a spec to specifically show that overriding is possible, as opposed to it being implied.

Is this more of what you mean?

```ruby
# custom_transaction_form.rb
class CustomTransactionForm < YAAF::Form
  delegate :transaction, to: User
end
```

```ruby
# form_spec.rb
RSpec.describe YAAF::Form do
  describe '#transaction' do
    it 'is delegated to ActiveRecord::Base' do
      form = YAAF::Form.new

      expect(ActiveRecord::Base).to receive(:transaction).and_call_original

      form.transaction { User.count }
    end

    it 'can be overridden by a subclass' do
      custom_form = CustomTransactionForm.new

      expect(User).to receive(:transaction).and_call_original
      expect(ActiveRecord::Base).not_to receive(:transaction).and_call_original

      custom_form.transaction { User.count }
    end
  end
end
```

Comment (santib) [reply spec/yaaf/form_spec.rb:3]: exactly, it could help to find regressions 😄

Comment (thewatts) [reply spec/yaaf/form_spec.rb]: 💪 perfect - just amended the original commit and pushed.

Thanks!

**Review by martinjaimem — APPROVED** [review]

**Pull request merged**